h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostTest.java

```diff
@@ -2152,11 +2155,63 @@ public void testXGBoostFeatureInteractions() {
       assertEquals(overallFeatureInteractionsTable[1].length, 1);
       assertEquals(overallFeatureInteractionsTable[2].length, 7);
       
+      GBMModel.GBMParameters gbmParms = new GBMModel.GBMParameters();
+      gbmParms._train = parms._train;
+      gbmParms._response_column = parms._response_column;
+      gbmParms._ignored_columns = parms._ignored_columns;
+      gbmParms._seed = parms._seed;
+      gbmParms._build_tree_one_node = parms._build_tree_one_node;
+
+      GBMModel gbmModel = new GBM(gbmParms).trainModel().get();
+      Scope.track_generic(gbmModel);
+      
+      // rank of GBM's and XGB's feature interactions sorted by gain should be similar:
+      double level0avgDistance = calculateAverageDistanceOfSortedInteractions(gbmModel.getFeatureInteractions(0,100,-1), model.getFeatureInteractions(0,100,-1));
+      // the bigger is the interaction depth, the bigger is the difference in XGB vs. GBM trees:
+      double level1avgDistance = calculateAverageDistanceOfSortedInteractions(gbmModel.getFeatureInteractions(1,100,-1), model.getFeatureInteractions(1,100,-1));
+      double level2avgDistance = calculateAverageDistanceOfSortedInteractions(gbmModel.getFeatureInteractions(2,100,-1), model.getFeatureInteractions(2,100,-1));
+
+      assertEquals(level0avgDistance, 0.28571428, 0.000001);
+      assertEquals(level1avgDistance, 4.06896551, 0.000001);
+      assertEquals(level2avgDistance, 19.3181818, 0.000001);
+      
     } finally {
       Scope.exit();
     }
   }
+  
+  // if some interaction is not present in both inputs, it is ignored
+  private static double calculateAverageDistanceOfSortedInteractions(FeatureInteractions featureInteractions1, FeatureInteractions featureInteractions2) {
+    List<KeyValue> list1 = new ArrayList<>();
+    List<KeyValue> list2 = new ArrayList<>();
+
+    for (Map.Entry<String, FeatureInteraction> featureInteraction : featureInteractions1.entrySet()) {
+      list1.add(new KeyValue(featureInteraction.getKey(), featureInteraction.getValue().gain));
+    }
+    list1.sort((a,b) -> a.getValue() < b.getValue() ? -1 : a.getValue() == b.getValue() ? 0 : 1);
 
+    for (Map.Entry<String, FeatureInteraction> featureInteraction : featureInteractions2.entrySet()) {
+      list2.add(new KeyValue(featureInteraction.getKey(), featureInteraction.getValue().gain));
+    }
+    list2.sort((a,b) -> a.getValue() < b.getValue() ? -1 : a.getValue() == b.getValue() ? 0 : 1);
+
+    List<String> sortedKeys1 = list1.stream().map(KeyValue::getKey).collect(Collectors.toList());
```

Comment:
sorting could be merged into the stream().sorted(Comparator.comparing().map()...